working-groups/meta/rfc-drafts/compiler-team-contributors.md

```diff
@@ -0,0 +1,244 @@
+variety of tasks that need to be done to support the compiler and
+which should make one eligible for membership. Suchs tasks would
+include organizing meetings, participating in meetings, bisecting and
+triaging issues, writing documentation, working on the
+rustc-guide. The most important criteria for elevation to contributor,
+in particular, is **regular and consistent** participation. The most
+important criteria for elevation to member is **actively shaping the
+direction of the team or compiler**.
+
+## Alumni status
+
+If at any time a current contributor or member wishes to take a break
+from participating, they can opt to put themselves into alumni status.
+When in alumni status, they will be removed from Github aliases and
+the like, so that they need not be bothered with pings and messages.
+They will also not have r+ privileges. **Alumni members will however
+still remain members of the GitHub org overall.**
+
+People in alumni status can ask to return to "active" status at any
+time.
+
+People in alumni status are still members of the team at the level
+they previously attained and they may publicly indicate that, though
+they should indicate the time period for which they were active as
+well.
+
+## Automatic alumni status after 6 months of inactivity
+
+If a contributor or a member has been inactive in the compiler for 6
+months, then we will ask them if they would like to go to alumni
+status. If they respond yes or do not respond, they can be placed on
+alumni status.  If they would prefer to remain active, that is also
+fine, but they will get asked again periodically if they continue to
+be inactive.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Why should we *not* do this?
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This RFC represents, effectively, the smallest extension to our structure
+that could possibly work. One could imagine more elaborate structures along a few dimensions.
+
+**More senior levels of membership.** One concern is that the set of
+**members** of the compiler team may grow too large for things like
+FCP (where each person must check their box) to be feasible. This
+could be resolved by moving away from FCP-based decision making (which
+is rarely used in the compiler anyhow), but it may also be worth
+considering another level of membership (e.g., a **senior
+member"). Senior members could be used for FCP-level decisions, which
+would presumably be relatively rare. At present there is a kind of
+implicit amount of "seniority" amongst members, where the opinions of
+people who have been around for longer are obviously given great
+weight, but formalizing this could have value.
+
+**Specialists and organizers.** Right now, we don't draw a distinction
+between people who write code and those who (for example) perform more
+organizational roles (as of the time of this writing, we don't have
+any members who perform more organizational roles exclusively, but
```

Comment:
(repeating myself from the meeting so it's written down here) 

I think that something we could do initially that isn't too much work would be to support some sort of "subtitle" that a member could decide on that would be displayed against their name wherever their membership is listed. This could include whatever the member likes, but it could be used to describe areas they focus on and how they're involved.

It could be "Compiler documentation", "Performance and optimization", "Async/await and generators", "Documenting compiler team practices and general contribution", and so on.

Reply:
I really like this idea, yes.

Reply:
Filed as #64